### PR TITLE
Release src/js as a npm package

### PR DESCRIPTION
**Pull request by rth**
Changes to be able to release of src/js as a standalone NPM package  (https://www.npmjs.com/package/pyodide).

Closes https://github.com/pyodide/pyodide/issues/328
xref: https://github.com/pyodide/pyodide/issues/14

Will add more documentation for using Pyodide in Node (https://github.com/pyodide/pyodide/issues/1725) in a follow up PR.

### CODE REVIEW

File: src/core/pyproxy.js [hunk 1]
```diff
@@ -17,7 +17,7 @@
  * See Makefile recipe for src/js/pyproxy.js
  */
 
-import { Module } from "../js/module.js";
+import { Module } from "./module.js";
```
Comment (hoodmane) [inline src/core/pyproxy.js:20]: This will mess up the PyProxy API docs.

Comment (hoodmane) [reply src/core/pyproxy.js:20]: I guess we can make a symlink `pyproxy.js` ==> `../core/pyproxy.gen.js`.

Comment (hoodmane) [reply src/core/pyproxy.js:20]: Actually the output it doesn't seem to be messed up so I guess this change is fine by itself.

File: src/js/README.md [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
 
-The Javascript code in this folder is responsible for the following tasks:
+<a href="https://www.npmjs.com/package/pyodide"><img src="https://img.shields.io/npm/v/pyodide" alt="npm"></a>
 
-1. Defines the public Javascript API
+## Usage
+
+Download and extract Pyodide packages from [Github
+releases](https://github.com/pyodide/pyodide/releases)
+(`pyodide-build-*.tar.bz2`). The version of the release needs to match exactly the version of this package.
```
Comment (hoodmane) [inline src/js/README.md:9]: We should try to simplify this by copying the `build` directory into `src/js` in the `prepublish` script.

Comment (rth) [reply src/js/README.md:9]: I agree that we need to simplify, but as discussed in https://github.com/pyodide/pyodide/issues/14#issuecomment-875116949 I'm not convinced that bundling all packages is the solution. We need a solution that would also work for micropip installed packages anyway. 

So I'm planning to add some local caching next and support remote CDN in node so there is no difference between browser and node usage when starting Pyodide.

**Comment by rth** [comment]
Merging. We can improve this in follow up PRs.

**Pull request merged**